workloads/wmt/wmt_jax/workload.py

```diff
@@ -0,0 +1,250 @@
+        num_heads=self.config.num_heads,
+        num_layers=self.config.num_layers,
+        qkv_dim=self.config.qkv_dim,
+        mlp_dim=self.config.mlp_dim,
+        max_len=max(self.config.max_target_length,
+                    self.config.max_eval_target_length),
+        dropout_rate=self.config.dropout_rate,
+        attention_dropout_rate=self.config.attention_dropout_rate,
+        deterministic=False,
+        decode=False,
+        kernel_init=nn.initializers.xavier_uniform(),
+        bias_init=nn.initializers.normal(stddev=1e-6))
+    self.eval_config = self.train_config.replace(deterministic=True)
+    self.predict_config = self.train_config.replace(
+        deterministic=True, decode=True)
+    self.p_eval_step = jax.pmap(
+        functools.partial(train.eval_step, config=self.eval_config),
+        axis_name="batch")
+    self.p_init_cache = jax.pmap(
+        functools.partial(
+            train.initialize_cache,
+            max_decode_len=self.config.max_predict_length,
+            config=self.predict_config),
+        axis_name="batch")
+    self.p_pred_step = jax.pmap(
+        functools.partial(
+            train.predict_step,
+            config=self.predict_config,
+            beam_size=self.config.beam_size),
+        axis_name="batch",
+        static_broadcasted_argnums=(3, 4))  # eos token, max_length are constant
+
+    rng, init_rng = jax.random.split(rng)
+    input_shape = (self.config.per_device_batch_size,
+                   self.config.max_target_length)
+    target_shape = (self.config.per_device_batch_size,
+                    self.config.max_target_length)
+
+    initial_variables = jax.jit(models.Transformer(self.eval_config).init)(
+        init_rng, jnp.ones(input_shape, jnp.float32),
+        jnp.ones(target_shape, jnp.float32))
+
+    initial_params = initial_variables["params"]
+
+    return initial_params, None
+
+  def model_fn(
+      self, params: spec.ParameterContainer,
+      augmented_and_preprocessed_input_batch: spec.Tensor,
+      model_state: spec.ModelAuxiliaryState, mode: spec.ForwardPassMode,
+      rng: spec.RandomState,
+      update_batch_norm: bool) -> Tuple[spec.Tensor, spec.ModelAuxiliaryState]:
+    del model_state
+    del rng
+    del update_batch_norm
+
+    model_config = self.train_config if mode == spec.ForwardPassMode.TRAIN else self.eval_config
+    inputs, targets = augmented_and_preprocessed_input_batch[
+        "inputs"], augmented_and_preprocessed_input_batch["targets"]
+    logits_batch = models.Transformer(model_config).apply({"params": params},
+                                                          inputs, targets)
+
+    return logits_batch, None
+
+  def loss_fn(
+      self,
+      label_batch: spec.Tensor,  # Dense (not one-hot) labels.
+      logits_batch: spec.Tensor) -> spec.Tensor:
+
+    weights = jnp.where(label_batch > 0, 1.0, 0.0)
+    vocab_size = logits_batch.shape[-1]
+    confidence = 1.0 - self.config.label_smoothing
```

Comment:
I think we should refactor this to be applied inside `submission.py`, because this is a tunable regularization hyperparameter that the submitter will be providing, so ideally we won't have any submission-specific logic inside `workload.py`. I think this should be pretty easy to do by just applying the label smoothing in `def loss_fn(params): ...` inside `train_step`.